src/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_comm.py

```diff
@@ -934,7 +934,7 @@ def internal_change_variable_json(py_db, request):
         )
         return
 
-    child_var = variable.change_variable(arguments.name, arguments.value, py_db, fmt=fmt)
+    child_var = variable.change_variable(arguments.name, arguments.value, py_db, fmt=fmt, scope=scope)
```

Comment:
This is where the setVariable request is handled. I added the scope to the change_variable